meadow-client/src/Data/BigInteger.purs

```diff
@@ -0,0 +1,54 @@
+module Data.BigInteger (BigInteger, fromInt, fromString) where
+
+import Prelude
+
+import Data.BigInt (BigInt, toString)
+import Data.Generic.Rep (class Generic)
+import Data.Generic.Rep.Eq (genericEq)
+import Data.Generic.Rep.Ord (genericCompare)
+import Data.Maybe (Maybe)
+import Data.Newtype (class Newtype, unwrap)
+import Test.QuickCheck (class Arbitrary, arbitrary)
+
+import Data.BigInt as BigInt
+
+newtype BigInteger
+  = BigInteger BigInt
+
+derive instance newtypeBigInteger :: Newtype BigInteger _
+
+derive instance genericBigInteger :: Generic BigInteger _
+
+instance eqBigInteger :: Eq BigInteger where
+  eq = genericEq
+
+instance ordBigInteger :: Ord BigInteger where
+  compare v = genericCompare v
+
+instance showBigInteger :: Show BigInteger where
+  show = toString <<< unwrap
+
+instance arbitraryBigInteger :: Arbitrary BigInteger where
+  arbitrary = map fromInt arbitrary
+
+fromInt :: Int -> BigInteger
+fromInt = BigInteger <<< BigInt.fromInt
+
+fromString :: String -> Maybe BigInteger
+fromString s = BigInteger <$> BigInt.fromString s
+
+instance semiringBigInteger :: Semiring BigInteger where
+  add a b = add a b
```

Comment:
Surely this is an infinite loop - you haven't unpacked the newtype...?

Reply:
Same with a few other calls in here.

Reply:
oops